src/Core/Grid/Query/CartQueryBuilder.php

```diff
@@ -0,0 +1,222 @@
+    ) {
+        parent::__construct($connection, $dbPrefix);
+        $this->criteriaApplicator = $criteriaApplicator;
+        $this->contextShopIds = $contextShopIds;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getSearchQueryBuilder(SearchCriteriaInterface $searchCriteria)
+    {
+        $dayInSeconds = 86400; // 24h
+
+        $qb = $this->getBaseQuery($searchCriteria->getFilters());
+        $qb->addSelect('SQL_CALC_FOUND_ROWS c.id_cart, c.date_add');
+        $qb->addSelect('
+            IF (IFNULL(o.id_order, "not_ordered") = "not_ordered",
+                IF(TIME_TO_SEC(TIMEDIFF(:current_date, c.date_add)) > :day_in_seconds,
+                 "abandoned_cart", "not_ordered"),
+                    o.id_order) AS status
+        ');
+        $qb->addSelect('CONCAT(LEFT(cu.firstname, 1), ". ", cu.lastname) AS customer_name');
+        $qb->addSelect('ca.name AS carrier_name, c.date_add, IF(con.id_guest, 1, 0) id_guest');
+        $qb->setParameter('day_in_seconds', $dayInSeconds);
+
+        $this->criteriaApplicator->applyPagination($searchCriteria, $qb);
+
+        $this->applySorting($qb, $searchCriteria);
+
+        return $qb;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getCountQueryBuilder(SearchCriteriaInterface $searchCriteria)
+    {
+        $qb = $this->connection->createQueryBuilder();
+        $qb->select('FOUND_ROWS()');
+
+        return $qb;
+    }
+
+    /**
+     * @param array $filters
+     *
+     * @return QueryBuilder
+     */
+    private function getBaseQuery(array $filters)
+    {
+        $halfHourInSeconds = 1800; // 0.5h
+
+        $subSql = $this->connection->createQueryBuilder()
+            ->select('co.id_guest')
+            ->from($this->dbPrefix . 'connections', 'co')
+            ->where('TIME_TO_SEC(TIMEDIFF(:current_date, co.date_add)) < :half_hour_in_seconds')
+            ->setMaxResults(1)
+        ;
+
+        $qb = $this->connection->createQueryBuilder()
+            ->from($this->dbPrefix . 'cart', 'c')
+            ->leftJoin('c', $this->dbPrefix . 'customer', 'cu', 'cu.id_customer = c.id_customer')
+            ->leftJoin('c', $this->dbPrefix . 'carrier', 'ca', 'ca.id_carrier = c.id_carrier')
+            ->leftJoin('c', $this->dbPrefix . 'currency', 'cr', 'cr.id_currency = c.id_currency')
+            ->leftJoin('c', $this->dbPrefix . 'orders', 'o', 'o.id_cart = c.id_cart')
+            ->leftJoin('c', '(' . $subSql->getSQL() . ')', 'con', 'con.id_guest = c.id_guest')
+            ->leftJoin('c', $this->dbPrefix . 'shop', 's', 's.id_shop = c.id_shop')
+            ->andWhere('c.id_shop IN (:context_shop_ids)')
+            ->setParameter('current_date', date('Y-m-d H:i:s'))
+            ->setParameter('context_shop_ids', $this->contextShopIds, Connection::PARAM_INT_ARRAY)
+            ->setParameter('half_hour_in_seconds', $halfHourInSeconds)
+        ;
+
+        $strictComparisonFilters = [
+            'status' => 'o.id_order',
```

Comment:
```suggestion
            'status' => 'status,
```

I'm not sure if this works but we should try, because as it is you can only search by order id, meaning you can't filter cart that have not been ordered based on the status that is constructed in the request (L74)

Reply:
I modified the handler so we can either search by order id or by status